lib/packagekit-glib2/pk-progress.c

```diff
@@ -774,6 +779,51 @@ pk_progress_get_uid (PkProgress *progress)
 	return progress->priv->uid;
 }
 
+/**
+ * pk_progress_set_sender:
+ * @progress: a valid #PkProgress instance
+ * @name: a D-Bus name
+ *
+ * Set the D-Bus name of the client that started this transaction.
+ *
+ * Return value: %TRUE if value changed.
+ *
+ * Since: 1.2.6
+ **/
+gboolean
+pk_progress_set_sender (PkProgress *progress, const gchar *name)
+{
+	g_return_val_if_fail (PK_IS_PROGRESS (progress), FALSE);
+
+	/* the same as before? */
+	if (progress->priv->sender == name)
+		return FALSE;
+
+	/* new value */
+	g_free (progress->priv->sender);
+	progress->priv->sender = g_strdup (name);
+	g_object_notify (G_OBJECT(progress), "sender");
+
+	return TRUE;
+}
+
+/**
+ * pk_progress_get_sender:
+ * @progress: a valid #PkProgress instance
+ *
+ * Get the D-Bus name of the client that started this transaction.
+ *
+ * Return value: a unique D-Bus name
```

Comment:
Probably safest to just call it a “D-Bus name”, as it could theoretically be a well-known _or_ a unique one. Although in practice it’ll always be a unique one